util/metrics.go

```diff
@@ -0,0 +1,115 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package util
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const metricsNamespace = "epr"
+
+var (
+	httpInFlightRequests = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: metricsNamespace,
+		Name:      "in_flight_requests",
+		Help:      "A gauge of requests currently being served by the http server.",
+	})
+
+	httpRequestsTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: metricsNamespace,
+			Name:      "http_requests_total",
+			Help:      "A counter for requests to the http server.",
+		},
+		[]string{"code", "method", "path"},
+	)
+
+	httpRequestDurationSeconds = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metricsNamespace,
+			Name:      "http_request_duration_seconds",
+			Help:      "A histogram of latencies for requests to the http server.",
+		},
+		[]string{"code", "method", "path"},
+	)
+
+	httpRequestSizeBytes = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metricsNamespace,
+			Name:      "http_request_size_bytes",
+			Help:      "A histogram of sizes of requests to the http server.",
+			Buckets: []float64{
+				16,
+				32,
+				64,
+				128,
+				256,
+				512,
+				1024,             /* 1KiB */
+				64 * 1024,        /* 64KiB */
+				256 * 1024,       /* 256KiB */
+				1024 * 1024,      /* 1MiB */
+				64 * 1024 * 1024, /* 64MiB */
+			},
+		},
+		[]string{"code", "method", "path"},
+	)
+
+	httpResponseSizeBytes = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metricsNamespace,
+			Name:      "http_response_size_bytes",
+			Help:      "A histogram of response sizes for requests to the http server.",
+			Buckets: []float64{
+				10,
+				64,
+				256,
+				512,
+				1024,              /* 1KiB */
+				64 * 1024,         /* 64KiB */
+				256 * 1024,        /* 256KiB */
+				512 * 1024,        /* 512KiB */
+				1024 * 1024,       /* 1MiB */
+				64 * 1024 * 1024,  /* 64MiB */
+				512 * 1024 * 1024, /* 512MiB */
```

Comment:
Is there any reasoning behind the selection of these buckets? Maybe we can use the current APM data we collect now to select values according to our use case.

Reply:
No reason for those buckets, I've set them just randomly and I was checking how they fit executing some curls.

If APM exposes that data so it can be checked the actual values, I'll check it and update this list.

Reply:
With v2 changes, we don't expect any responses great than a few KBs. In case the Fleet wants to download a ZIP file, it will be redirected (HTTP 301) to the https://package-storage.elastic.co/.